### PR TITLE
[DOCU 2411] and [DOCU-2444] Plugin Overviews

### PR DESCRIPTION
**Pull request by cloudjumpercat**
### Summary

* (Kong Plugins: Overview) Explains how Kong uses plugins (Move and update overview from hub: https://docs.konghq.com/hub/plugins/overview/)
* (Understanding Kong: Plugins) New doc: Provides an overview of what a plugin is and how they work in Gateway
### Reason

https://konghq.atlassian.net/browse/DOCU-2444
https://konghq.atlassian.net/browse/DOCU-2411
### Testing

Does it work to reuse this content in the Plugin Hub Overview, Understanding Kong: Plugins, and Kong Plugins: Overview?

### CODE REVIEW

**Comment by Guaris** [comment]
I think we should leave the text in `app/_hub/plugins/overview/_index.md` even if we are using it and adapting it to the other doc. It may make sense to use an include here so we have one source of truth.

File: app/_includes/md/about-plugins.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+Plugins consist of modules interacting with the request/response objects or
+streams via a Plugin Development Kit (or PDK) to implement arbitrary logic.
+Kong provides PDKs for two languages: Lua and Go. Both of these PDKs are sets
+of functions that a plugin can use to facilitate interactions between plugins
+and the core (or other components) of Kong.
+
+To start creating your own plugins, check out the PDK documentation:
+* [Plugin Development Guide](/gateway/latest/plugin-development) and the
+[Plugin Development Kit reference](/gateway/latest/pdk)
```
Comment (Guaris) [inline app/_includes/md/about-plugins.md:97]:
```suggestion
* [Plugin Development Kit reference](/gateway/latest/pdk)
```

File: src/gateway/understanding-kong/key-concepts/plugins.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
-title: plugins
-
+title: Plugins
+concept_type: explanation
 ---
 
-## PLACEHOLDER plugins
+{% include_cached /md/about-plugins.md %}
+
+## See also
```
Comment (Guaris) [inline src/gateway/understanding-kong/key-concepts/plugins.md:8]: Adding links to the other key-concepts docs here would be a good idea. Maybe a section titled ##Other key concepts ?

**Comment by Guaris** [comment]
Nice use of an include

File: app/_includes/md/about-plugins.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+**Route**
+: The Kong entity representing a way to map downstream requests to upstream
+services.
+
+**Consumer**
+: An entity that makes requests for Kong to proxy. It represents either a user
+or an external service.
+
+**Credential**
+: A unique string associated with a Consumer; also referred to as an API key.
```
Comment (Guaris) [inline app/_includes/md/about-plugins.md:55]:
```suggestion
: The API key. A unique string associated with a consumer. 
```

File: app/_includes/md/about-plugins.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+with the correct version of Kong.
+
+### Third-party plugins
+
+Because third-party plugins are not maintained by Kong and are not bundled with
+the {{site.base_gateway}}, version compatibility is a bigger concern. See each
+individual plugin's page for its tested compatibility.
+
+If the versions on the plugin page are outdated, contact the maintainer through
+the Support link in the plugin documentation's sidebar.
```
Comment (Guaris) [inline app/_includes/md/about-plugins.md:82]:
```suggestion
the support link in the plugin documentation's sidebar.
```

Comment (Guaris) [reply app/_includes/md/about-plugins.md:82]: Can we just link it here?

**Review by Guaris — CHANGES_REQUESTED** [review]

File: src/gateway/understanding-kong/key-concepts/plugins.md (hunk 2 above)
Comment (cloudjumpercat) [reply src/gateway/understanding-kong/key-concepts/plugins.md:8]: @Guaris What other key-concept docs were you thinking of? In the include file, the last few sections already contain links to the PDK, plugin development guide, plugin hub contribution guidelines, and plugin other language support, so I wouldn't want to repeat those. I can probably add this Plugin Hub link to the include file so it's in all the "plugin overview" topics.

Comment (lena-larionova) [reply src/gateway/understanding-kong/key-concepts/plugins.md:8]: We also have this page: https://docs.konghq.com/konnect-platform/key-concepts/

It's not really well-named; at this point, we could rename it to release lifecycle terms or something similar. It used to have more terms but these are all that's left now.

Comment (Guaris) [reply src/gateway/understanding-kong/key-concepts/plugins.md:8]: <google-sheets-html-origin><style type="text/css"></style>

Key concepts |   |   |  
-- | -- | -- | --
  | Services |   | explanation
  | Routes |   | explanation
  | Upstreams |   | explanation
  | Plugins |   | explanation

The services, routes, and upstreams doc.

**Review by Guaris — APPROVED** [review]

**Pull request merged**